### PR TITLE
[CAS-482] retry message

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
[CAS-428](https://stream-io.atlassian.net/browse/CAS-428)

### Description

Adding message options for messages that failed for sync. 

A user will see this:

Screenshot_20201223-204046

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes. **No need**
- [ ] New code is covered by unit tests. **No need**
- [x] Comparison screenshots added for visual changes.
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/options/MessageOptionsView.kt [hunk 1]
```diff
@@ -27,34 +28,82 @@ public class MessageOptionsView : FrameLayout {
...
-        binding.threadReplyTV.configureListItem(configuration.threadReplyText, configuration.threadReplyIcon, iconsTint)
+        // Todo: Uncomment once reply is done.
+        // binding.replyTV.configureListItem(configuration.replyText, configuration.replyIcon, iconsTint)
 
-        if (configuration.copyTextEnabled) {
-            binding.copyTV.isVisible = true
-            binding.copyTV.configureListItem(configuration.copyText, configuration.copyIcon, iconsTint)
-        } else {
-            binding.copyTV.isVisible = false
+        if (syncStatus == SyncStatus.COMPLETED) {
```
Comment (zsmb13) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/options/MessageOptionsView.kt:45]: Can _their_ message be in any another sync status somehow?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/options/MessageOptionsView.kt [hunk 2]
```diff
@@ -27,34 +28,82 @@ public class MessageOptionsView : FrameLayout {
...
+            }
+            SyncStatus.COMPLETED -> {
+                binding.threadReplyTV.configureListItem(
+                    configuration.threadReplyText,
+                    configuration.threadReplyIcon,
+                    iconsTint
+                )
+            }
+            SyncStatus.SYNC_NEEDED, SyncStatus.IN_PROGRESS -> {
+            }
```
Comment (zsmb13) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/options/MessageOptionsView.kt:86]: What happens to the "thread reply" option in this case?

**Review by zsmb13 — COMMENTED** [review]
Nice work, the structure around these parts is making it easy to extend and review this 👍 

Got just two small questions about some cases that I'm unsure about.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/options/MessageOptionsView.kt (hunk 1 above)
Comment (leandroBorgesFerreira) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/options/MessageOptionsView.kt:45]: Indeed, COMPLETED is the only possible way.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/options/MessageOptionsView.kt (hunk 2 above)
Comment (leandroBorgesFerreira) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/options/MessageOptionsView.kt:86]: It becomes not visible, but this is made in the XML in this situation. I'll do also move it to code.

**Comment by zsmb13** [comment]
Addressed my own comments + resolved conflicts with `develop` to keep this moving forward

**Review by ogkuzmin — APPROVED** [review]

**Pull request merged**